oci/table_oci_core_instance_configuration.go

```diff
@@ -0,0 +1,304 @@
+			},
+			{
+				Name:        "time_created",
+				Description: "The date and time the instance configuration was created.",
+				Type:        proto.ColumnType_TIMESTAMP,
+				Transform:   transform.FromField("TimeCreated.Time"),
+			},
+			{
+				Name:        "deferred_fields",
+				Description: "Parameters that were not specified when the instance configuration was created, but that are required to launch an instance from the instance configuration.",
+				Type:        proto.ColumnType_JSON,
+				Hydrate:     getInstanceConfiguration,
+			},
+			{
+				Name:        "instance_details",
+				Description: "The intance configuration details.",
+				Type:        proto.ColumnType_JSON,
+				Hydrate:     getInstanceConfiguration,
+			},
+
+			// tags
+			{
+				Name:        "defined_tags",
+				Description: ColumnDescriptionDefinedTags,
+				Type:        proto.ColumnType_JSON,
+			},
+			{
+				Name:        "freeform_tags",
+				Description: ColumnDescriptionFreefromTags,
+				Type:        proto.ColumnType_JSON,
+			},
+
+			// Standard Steampipe columns
+			{
+				Name:        "tags",
+				Description: ColumnDescriptionTags,
+				Type:        proto.ColumnType_JSON,
+				Transform:   transform.From(instanceConfigurationTags),
+			},
+			{
+				Name:        "title",
+				Description: ColumnDescriptionTitle,
+				Type:        proto.ColumnType_STRING,
+				Transform:   transform.FromField("DisplayName"),
+			},
+
+			// Standard OCI columns
+			{
+				Name:        "region",
+				Description: ColumnDescriptionRegion,
+				Type:        proto.ColumnType_STRING,
+				Transform:   transform.FromCamel().Transform(extractInstanceConfigurationRegion),
+			},
+			{
+				Name:        "compartment_id",
+				Description: ColumnDescriptionCompartment,
+				Type:        proto.ColumnType_STRING,
+				Transform:   transform.FromField("CompartmentId"),
+			},
+			{
+				Name:        "tenant_id",
+				Description: ColumnDescriptionTenant,
+				Type:        proto.ColumnType_STRING,
+				Hydrate:     plugin.HydrateFunc(getTenantId).WithCache(),
+				Transform:   transform.FromValue(),
+			},
+		},
+	}
+}
+
+//// LIST FUNCTION
+
+func listCoreInstanceConfigurations(ctx context.Context, d *plugin.QueryData, _ *plugin.HydrateData) (interface{}, error) {
+	logger := plugin.Logger(ctx)
+	region := plugin.GetMatrixItem(ctx)[matrixKeyRegion].(string)
+	compartment := plugin.GetMatrixItem(ctx)[matrixKeyCompartment].(string)
+
+	// Create Session
+	session, err := coreComputeManagementService(ctx, d, region)
+	if err != nil {
+		logger.Error("oci_core_instance_configuration.listCoreInstanceConfigurations", "connection_error", err)
+		return nil, err
+	}
+
+	equalQuals := d.KeyColumnQuals
+
+	// Return nil, if given compartment_id doesn't match
+	if equalQuals["compartment_id"] != nil && compartment != equalQuals["compartment_id"].GetStringValue() {
+		return nil, nil
+	}
+
+	// Build request parameters
+	// request := buildCoreInstanceFilters(equalQuals)
```

Comment:
please remove commented code if not required anymore

Reply:
Removed the unnecessary commented codes.